packages/main/src/MonthPicker.ts

```diff
@@ -224,13 +221,7 @@ class MonthPicker extends CalendarPart implements ICalendarPicker {
 	}
 
 	_onHomeOrEnd(homePressed: boolean) {
-		this._months.forEach(row => {
-			const indexInRow = row.findIndex(item => CalendarDate.fromTimestamp(parseInt(item.timestamp) * 1000).getMonth() === this._calendarDate.getMonth());
-			if (indexInRow !== -1) { // The current month is on this row
-				const index = homePressed ? 0 : ROW_SIZE - 1; // select the first (if Home) or last (if End) month on the row
-				this._setTimestamp(parseInt(row[index].timestamp));
-			}
-		});
+		this._setTimestamp(parseInt(this._months[homePressed ? 0 : PAGE_SIZE - 1].timestamp));
```

Comment:
According to the keyboard handling spec:

> Home: Moves focus to the first column of the same row and same page.
> End: Moves focus to the last column of the same row and same page. Decouples focus from selection.

Currently the Home/End moves focus from the first grid cells to the last grid cell thus changing the current row.

Additionally Home/End combination doesn't work if the primary calendar type is different than Gregorian.
Not that it worked before. We could log separate incident for those keyboard interactions.